frontend/controller/actions/group.js

```diff
@@ -131,6 +181,101 @@ export default (sbp('sbp/selectors/register', {
   'gi.actions/group/switch': function (groupId) {
     sbp('state/vuex/commit', 'setCurrentGroupId', groupId)
   },
+  'gi.actions/group/addChatRoom': async function (params: GIActionParams) {
+    const message = await sbp('gi.actions/chatroom/create', { data: params.data })
+
+    await sbp('chelonia/out/actionEncrypted', {
+      ...params,
+      action: 'gi.contracts/group/addChatRoom',
+      data: {
+        ...params.data,
+        chatRoomID: message.contractID()
+      }
+    })
+
+    return message
+  },
+  'gi.actions/group/joinChatRoom': async function (params: GIActionParams) {
+    try {
+      const rootState = sbp('state/vuex/state')
+      const username = params.data.username || rootState.loggedIn.username
+      await sbp('gi.actions/chatroom/join', {
+        contractID: params.data.chatRoomID,
+        data: { username }
```

Comment:
Added hooks handler.
But regarding `publishOptions`, I can not see handling it except the actions which create something(group, mailbox, identity, ...). Actually, I have never used `publishOptions` so far.
Should I just send that options to the another action?

Reply:
You might not use `publishOptions`, but the caller of this action might. Therefore those should be passed down as well.

A shortcut is to include everything in the params except for `options` (since `options` is not for chelonia, but if you want to specify any special options for these actions), using `_.omit()` like I [showed here](https://github.com/okTurtles/group-income/pull/1143#discussion_r846834994), and then override any `data` as appropriate.

Reply:
Got it. I will check all the actions, not only this one...

Reply:
Checked all the actions in `group.js` and `chatroom.js` and replaced all the `...param` to `...omit(params, ['options'])`.
This should be done, right?